pillarbox-core-business/src/main/java/ch/srgssr/pillarbox/core/business/source/SpriteSheetMediaPeriod.kt

```diff
@@ -43,20 +49,25 @@ internal class SpriteSheetMediaPeriod(
         .build()
     private val tracks = TrackGroupArray(TrackGroup("sprite-sheet-srg", format))
     private var positionUs = 0L
-
+    private var currentLoadingJob: Job? = null
     override fun prepare(callback: MediaPeriod.Callback, positionUs: Long) {
-        callback.onPrepared(this)
         this.positionUs = positionUs
+        currentLoadingJob?.cancel()
         isLoading.set(true)
-        spriteSheetLoader.loadSpriteSheet(spriteSheet) { bitmap ->
-            this.bitmap = bitmap
+        bitmap = null
+        callback.onPrepared(this)
+        currentLoadingJob = MainScope().launch(coroutineContext) {
```

Comment:
According to the documentation, `MainScope` is used for UI components.
What about doing this instead?
```suggestion
        currentLoadingJob = CoroutineScope(coroutineContext).launch {
```

Reply:
I kept the scope we previously used, but indeed we could improve it.